src/generate_pisa_index.cpp

```diff
@@ -21,19 +41,138 @@ std::ostream &write_sequence(std::ostream &os, gsl::span<T> sequence)
 }
 
 
+struct inverted_index {
+
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_documents {};
+  std::unordered_map<uint32_t, std::vector<uint32_t>> m_frequencies {};
+  std::map<uint32_t, uint32_t> m_document_sizes {};
+  std::vector<std::string> m_plain_terms {};
+ 
+  void add_document_lengths(const std::string doclen_file) {
```

Comment:
Missed `&`, it's being unnecessarily copied right now.

Reply:
Thanks!